transaction/core/src/validation/validate.rs

```diff
@@ -30,6 +30,7 @@ use rand_core::{CryptoRng, RngCore};
 pub fn validate<R: RngCore + CryptoRng>(
     tx: &Tx,
     current_block_index: u64,
+    block_version: BlockVersion,
     root_proofs: &[TxOutMembershipProof],
     minimum_fee: u64,
     csprng: &mut R,
```

Comment:
it might be a good idea to have explicit checks in this function comparing `BlockVersion` with `BlockVersion::MAX`, even though it's supposed to be a static invariant. that is what we are doing in ledger db right now. lmk what you think about that.